reactions: Add sheet to view who reacted to a message.

chimnayajith opened the pull request:
## Description

This pull request introduces a **Reaction List** sheet which can be accessed from either the context menu or by long pressing the reaction chips. Currently the app shows who reacted when there are up to 3 of them. But once there are more, we just show counts, and the user may want to see specifically who left the reactions. 

## Related Issues

Closes #740  

## Screenshots

### Dark Mode
<div>
  <img src="https://github.com/user-attachments/assets/d99727a8-b15a-4ddd-9faf-87074b8c4bd5" width="250">
  <img src="https://github.com/user-attachments/assets/62298e77-873c-4e3a-b9e5-26773ccf5dab" width="250">
</div>

### Light Mode
<div>
  <img src="https://github.com/user-attachments/assets/48069397-b0a8-4bcc-9808-e1b0a752f364" width="250">
  <img src="https://github.com/user-attachments/assets/cb27818e-8210-48e8-b9fe-89109da9f084" width="250">
</div>

chrisbobbe commented:
Hi @chimnayajith, I've just updated the issue description to make one aspect of it more visible: https://github.com/zulip/zulip-flutter/issues/740#issuecomment-2568346506

chrisbobbe commented:
Thanks! This PR needs tests.

Also, please reorder the two commits, so the commit that adds the icon comes first. That way the icon will be available for the commit that adds the feature.

chimnayajith commented:
@chrisbobbe I've reordered the commits and written the tests for the same. Please take a look when. Thank you.

chrisbobbe reviewed (commented):
Thanks! Comments below from a partial review.

Also, a question for you 🙂: what happens if you're looking at the tab for a particular reaction, and then the last vote for that reaction is removed?

chimnayajith commented on lib/widgets/emoji_reaction.dart:
```diff
@@ -127,23 +130,36 @@ class ReactionChipsList extends StatelessWidget {
-      ).toList());
+        messageId: messageId,
+        reactionWithVotes: reactionVotes,
+        onLongPress:(context){
+          showReactionListSheet(
+            context,
+            reactionList: reactions.aggregated,
+            initialTabIndex: index,
+            );
+        }
```
But that would mean passing the entire `reactions` list to each `ReactionChip` which seems unnecessary.

chimnayajith replied:
Wouldn't it be better to pass the reaction to the `ReactionChip` rather than get the reactions using messageId like this? or is there a better way
```dart
final reactions = store.messages[messageId]?.reactions;
```

chimnayajith commented:
I’ve addressed all the requested changes and marked them as resolved.

As per the conversation [here](https://chat.zulip.org/#narrow/channel/516-mobile-dev-help/topic/view-reactions.20sheet), I’ve implemented the live UI using the `MessgeListView` getter .

As for the question:
> What happens if you're viewing the tab for a particular reaction, and then the last vote for that reaction is removed?

Currently, when the last vote for a reaction is removed, the corresponding tab disappears, and the view switches to the previous tab. If there are no remaining reactions, a message saying 'No reactions yet' is shown. Does this behavior seem appropriate?

gnprice commented:
Thanks @chimnayajith for picking this up!

This is a post-launch issue, and it looks like there's still substantial work to do in order to (a) get this matching all the details of the design in Figma, and (b) get the structure of the code up to our standards. So as part of [focusing on launch issues](https://chat.zulip.org/#narrow/channel/243-mobile-team/topic/Flutter.20beta.20timeline/near/2078556) until the launch is ready, let's put this PR on hold for now.
